### PR TITLE
Added a better pre-chat example

### PR DESCRIPTION
**Pull request by sfyz**
Made the following changes:

1. Added a top-level button for chat so that you don't have to go through the KB interface to get there
2. Moved the chat setup into its own class so that it can be accessed from multiple locations
3. Made the pre-chat form more full-bodied to illustrate a better use case scenario (e.g. it now includes entity mappings)
4. Verified that it works for both Java and Kotlin

### CODE REVIEW

**Comment by sfyz** [comment]
@rileymacdonald-sf Let me know what you think about this sample app update... Also, I'm not sure why some of the checks were not successful...? Thx!

File: Examples/SnapinsSDKExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/ChatLauncher.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                });
+    }
+
+    /**
+     * Configures pre-chat fields if pre-chat is enabled in settings
+     */
+    private void initializePreChatData() {
+
+        if (!preChatEnabled()) {
+            chatUserData = Collections.<ChatUserData>emptyList();
```
Comment (rileymacdonald-sf) [inline Examples/SnapinsSDKExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/ChatLauncher.java]: Nit: these data types can be inferred (maybe you've left this in for clarity).

File: Examples/SnapinsSDKExample/app/src/kotlinImplDebug/java/com/salesforce/snapinssdkexample/ChatLauncher.kt [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+import android.content.Context
+import android.view.inputmethod.EditorInfo
+import androidx.fragment.app.FragmentActivity
+import com.salesforce.android.chat.core.ChatConfiguration
+import com.salesforce.android.chat.core.model.ChatEntity
+import com.salesforce.android.chat.core.model.ChatEntityField
+import com.salesforce.android.chat.core.model.ChatUserData
+import com.salesforce.android.chat.ui.ChatUI
+import com.salesforce.android.chat.ui.ChatUIClient
+import com.salesforce.android.chat.ui.model.PreChatPickListField
```
Comment (rileymacdonald-sf) [inline Examples/SnapinsSDKExample/app/src/kotlinImplDebug/java/com/salesforce/snapinssdkexample/ChatLauncher.kt]: Nit: unused import.

File: Examples/SnapinsSDKExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/ChatLauncher.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
+package com.salesforce.snapinssdkexample;
+
+import com.salesforce.android.chat.core.ChatConfiguration;
+import com.salesforce.android.chat.core.model.ChatEntity;
+import com.salesforce.android.chat.core.model.ChatEntityField;
+import com.salesforce.android.chat.core.model.ChatUserData;
+import com.salesforce.android.chat.ui.ChatUI;
+import com.salesforce.android.chat.ui.ChatUIClient;
+import com.salesforce.android.chat.ui.model.PreChatPickListField;
```
Comment (rileymacdonald-sf) [inline Examples/SnapinsSDKExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/ChatLauncher.java]: Nit: unused import.

File: Examples/SnapinsSDKExample/build.gradle [hunk 4]
```diff
@@ -13,7 +13,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.3.1'
+        classpath 'com.android.tools.build:gradle:3.3.2'
```
Comment (rileymacdonald-sf) [inline Examples/SnapinsSDKExample/build.gradle]: Should probably create a task to update the build tools and kotlin version to the very latest (`3.4.0` and `1.3.30`).

File: Examples/SnapinsSDKExample/app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -55,16 +55,18 @@
...
+    <string name="knowledge_description_text">This feature gives you access to your org’s knowledge base. Once you point your app to your community URL with the right category group and root data
+                                              category, you can display your knowledge base to your users.</string>
+    <string name="chat_description_title">Chat</string>
+    <string name="chat_description_text">This feature gives you access to chat. Once you add the config settings to this app, you can chat with an agent or a chatbot.</string>
     <string name="authentication_description_title">Authentication</string>
-    <string name="authentication_description_text">Use Knowlegde or Cases as an authenticated or unauthenticated user</string>
+    <string name="authentication_description_text">Use Knowledge or Cases as an authenticated or unauthenticated user</string>
 
     <!-- Support FAB -->
     <string name="menu_case">Create Case</string>
```
Comment (rileymacdonald-sf) [inline Examples/SnapinsSDKExample/app/src/main/res/values/strings.xml]: Nit: Down ~lines 126 there's a bunch of `prechat_*` strings that appear to be unused now. Could probably remove them.

File: Examples/SnapinsSDKExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/SupportHomeViewAddition.java [hunk 6]
```diff
@@ -132,69 +132,8 @@ private Animator getAnimatorSet(View view, Float value) {
      * Configures and launches Live Agent Chat
      */
     private void launchChat() {
-        ChatConfiguration chatConfiguration;
```
Comment (rileymacdonald-sf) [inline Examples/SnapinsSDKExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/SupportHomeViewAddition.java]: Nit: this results in a bunch of unused imports.

File: Examples/SnapinsSDKExample/app/src/kotlinImplDebug/java/com/salesforce/snapinssdkexample/SupportHomeViewAddition.kt [hunk 7]
```diff
@@ -113,66 +113,8 @@ class SupportHomeViewAddition : KnowledgeViewAddition {
      * Configures and launches Live Agent Chat
      */
     private fun launchChat() {
-        // Create a UI configuration instance from a core config object
```
Comment (rileymacdonald-sf) [inline Examples/SnapinsSDKExample/app/src/kotlinImplDebug/java/com/salesforce/snapinssdkexample/SupportHomeViewAddition.kt]: Nit: this results in a bunch of unused imports.

**Comment by sfyz** [comment]
> Looks great! Good clarity on this. Runs good on API 21-28. Left a few nit picky comments, otherwise ready to go!

@rileymacdonald-sf Great feedback. Thanks! Incorporating now.

**Pull request merged**